src/engine/player/PlayerRig.ts

```diff
@@ -43,70 +48,80 @@ import { embodyAvatar } from "./embodyAvatar";
 import { addFlyControls } from "./FlyCharacterController";
 import { addKinematicControls } from "./KinematicCharacterController";
 import { addNametag } from "./nametags.game";
-import { Player, OurPlayer } from "./Player";
+import { Player } from "./Player";
+import { XRControllerComponent, XRHeadComponent, XRRayComponent } from "./XRComponents";
+import { createNetworkReplicator } from "../network/NetworkReplicator";
+import { transformCodec } from "../network/NetworkMessage";
+import { isHost } from "../network/network.common";
 
 const AVATAR_CAPSULE_HEIGHT = 1;
 const AVATAR_CAPSULE_RADIUS = 0.35;
 export const AVATAR_HEIGHT = AVATAR_CAPSULE_HEIGHT + AVATAR_CAPSULE_RADIUS * 2;
 const AVATAR_CAMERA_OFFSET = 0.06;
 
-export function registerPlayerPrefabs(ctx: GameContext) {
-  registerPrefab(ctx, {
-    name: "avatar",
-    type: PrefabType.Avatar,
-    create: (ctx: GameContext) => {
-      const physics = getModule(ctx, PhysicsModule);
-      const spawnPoints = spawnPointQuery(ctx.world);
-
-      const container = new RemoteNode(ctx.resourceManager);
-      const rig = createNodeFromGLTFURI(ctx, "/gltf/full-animation-rig.glb");
-
-      addChild(container, rig);
-
-      quat.fromEuler(rig.quaternion, 0, 180, 0);
+const avatarFactory = (ctx: GameContext) => {
```

Comment:
This looks great! Good proof that using the replicator API internally cleaned some things up.